packages/components/src/navigator/match-path.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * External dependencies
+ */
+import pathToRegexp from 'path-to-regexp';
+
+const cache = {};
+const cacheLimit = 10000;
+let cacheCount = 0;
+
+function compilePath( path, options ) {
+	const cacheKey = `${ options.end }${ options.strict }${ options.sensitive }`;
+	const pathCache = cache[ cacheKey ] || ( cache[ cacheKey ] = {} );
+
+	if ( pathCache[ path ] ) return pathCache[ path ];
+
+	const keys = [];
+	const regexp = pathToRegexp( path, keys, options );
+	const result = { keys, regexp };
+
+	if ( cacheCount < cacheLimit ) {
+		pathCache[ path ] = result;
+		cacheCount++;
+	}
+
+	return result;
+}
+
+/**
+ * Public API for matching a URL pathname to a path.
```

Comment:
The method is not public/exported outside the package right? Should we replace "Public API" with function?

Reply:
I'm actually considering removing this function/file for now. All what we really need for now is `path === currentPath` we can consider advanced use-cases later.